---
title: Lab 008 - Inference for rates.
author:
  - name: EPIB607 - Inferential Statistics
    affiliation: a
#  - name: Second Author
#    affiliation: a,b
address:
  - code: a
    address: McGill University
#  - code: b
#    address: Department of Neat Tricks, Whereever State University, Someplace, MC, 67890
lead_author_surname: Bhatnagar
#doi: "https://sahirbhatnagar.com/EPIB607/"
abstract: |
  In this exercise you will practice doing inference for one-sample rates. 
# Optional: Acknowledgements
#acknowledgements: |
#  [rticles](https://cran.r-project.org/package=rticles) package, and both packages rely on the
#  [PNAS LaTeX](http://www.pnas.org/site/authors/latex.xhtml) macros. Both these sources are
#  ([GPL-3](https://www.gnu.org/licenses/gpl-3.0.en.html) and
#  [LPPL (>= 1.3)](https://www.latex-project.org/lppl/)).
# Optional: One or more keywords
keywords:
  - Sampling distribution
  - Standard error
  - Normal distribution
  - Quantiles
  - Percentiles
  - Z-scores
papersize: letter
fontsize: 11pt
# Optional: Force one-column layout, default is two-column
one_column: true
# Optional: Enables lineno mode, but only if one_column mode is also true
#lineno: true
# Optional: Enable one-sided layout, default is two-sided
#one_sided: true
# Optional: Enable section numbering, default is unnumbered
numbersections: true
# Optional: Specify the depth of section number, default is 5
#secnumdepth: 5
# Optional: Skip inserting final break between acknowledgements, default is false
skip_final_break: true
# Optional: Bibliography 
bibliography: pinp
# Optional: Enable a 'Draft' watermark on the document
watermark: false
# footer_contents: "in-class exercise on rates"
output: pinp::pinp
# Required: Vignette metadata for inclusion in a package.
vignette: >
  %\VignetteIndexEntry{YourPackage-vignetteentry}
  %\VignetteKeywords{YourPackage, r, anotherkeyword}
  %\VignettePackage{Due Sepetember 28, 2018}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  chunk_output_type: console
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(message = FALSE, warning = FALSE, fig.width = 6, fig.height = 6)
library(dplyr)
library(knitr)
library(kableExtra)
library(mosaic)
library(tidyr)
```


\begin{table}[h]
\begin{center}
\begin{tabular}{|r|rr|rr|rr|}
\hline
$y$ & \multicolumn{2}{c}{95\%} & \multicolumn{2}{c}{90\%} & \multicolumn{2}{c}{80\%} \\
\hline
0 & 0.00 & 3.69 & 0.00 & 3.00 & 0.00 & 2.30 \\
1 & 0.03 & 5.57 & 0.05 & 4.74 & 0.11 & 3.89 \\
2 & 0.24 & 7.22 & 0.36 & 6.30 & 0.53 & 5.32 \\
3 & 0.62 & 8.77 & 0.82 & 7.75 & 1.10 & 6.68 \\
4 & 1.09 & 10.24 & 1.37 & 9.15 & 1.74 & 7.99 \\
& & & & & &  \\
5 & 1.62 & 11.67 & 1.97 & 10.51 & 2.43 & 9.27 \\
6 & 2.20 & 13.06 & 2.61 & 11.84 & 3.15 & 10.53 \\
7 & 2.81 & 14.42 & 3.29 & 13.15 & 3.89 & 11.77 \\
8 & 3.45 & 15.76 & 3.98 & 14.43 & 4.66 & 12.99 \\
9 & 4.12 & 17.08 & 4.70 & 15.71 & 5.43 & 14.21 \\
& & & & & &  \\
10 & 4.80 & 18.39 & 5.43 & 16.96 & 6.22 & 15.41 \\
11 & 5.49 & 19.68 & 6.17 & 18.21 & 7.02 & 16.60 \\
12 & 6.20 & 20.96 & 6.92 & 19.44 & 7.83 & 17.78 \\
13 & 6.92 & 22.23 & 7.69 & 20.67 & 8.65 & 18.96 \\
14 & 7.65 & 23.49 & 8.46 & 21.89 & 9.47 & 20.13 \\
& & & & & &  \\
15 & 8.40 & 24.74 & 9.25 & 23.10 & 10.30 & 21.29 \\
16 & 9.15 & 25.98 & 10.04 & 24.30 & 11.14 & 22.45 \\
17 & 9.90 & 27.22 & 10.83 & 25.50 & 11.98 & 23.61 \\
18 & 10.67 & 28.45 & 11.63 & 26.69 & 12.82 & 24.76 \\
19 & 11.44 & 29.67 & 12.44 & 27.88 & 13.67 & 25.90 \\
& & & & & &  \\
20 & 12.22 & 30.89 & 13.25 & 29.06 & 14.53 & 27.05 \\
21 & 13.00 & 32.10 & 14.07 & 30.24 & 15.38 & 28.18 \\
22 & 13.79 & 33.31 & 14.89 & 31.41 & 16.24 & 29.32 \\
23 & 14.58 & 34.51 & 15.72 & 32.59 & 17.11 & 30.45 \\
24 & 15.38 & 35.71 & 16.55 & 33.75 & 17.97 & 31.58 \\
30 & `r round(qgamma(0.025,30),2)` & `r round(qgamma(0.975,31),2)` & `r round(qgamma(0.05,30),2)` & `r round(qgamma(0.95,31),2)` & `r round(qgamma(0.10,30),2)` & `r round(qgamma(0.90,31),2)`\\
32 & `r round(qgamma(0.025,32),2)` & `r round(qgamma(0.975,33),2)` & `r round(qgamma(0.05,32),2)` & `r round(qgamma(0.95,33),2)` & `r round(qgamma(0.10,32),2)` & `r round(qgamma(0.90,33),2)`\\
33 & `r round(qgamma(0.025,33),2)` & `r round(qgamma(0.975,34),2)` & `r round(qgamma(0.05,33),2)` & `r round(qgamma(0.95,34),2)` & `r round(qgamma(0.10,33),2)` & `r round(qgamma(0.90,34),2)`\\
36 & `r round(qgamma(0.025,36),2)` & `r round(qgamma(0.975,37),2)` & `r round(qgamma(0.05,36),2)` & `r round(qgamma(0.95,37),2)` & `r round(qgamma(0.10,36),2)` & `r round(qgamma(0.90,37),2)`\\
\hline
\end{tabular}
\end{center}
\caption{Poisson based Confidence Intervals}
\end{table}


# Vacancy rates in the US Supreme Court 

Refer to the article _Updating a Classic: The Poisson Distribution and the Supreme Court Revisited_. 

a) What is the probability that a sitting President will appoint 3 judges? (Reflect on this given the current situation). 

b) Reproduce Table 1, i.e., determine how the **Probability** and **Expected** columns were calculated. 

c) Update Table 1 and calculate an updated vacancy rate (expressed as vacancies per year) for the period 1933-2020.

d) Based only on your updated point-estimate of the rate [i.e. without using any information on the health of the current court], what is the probability that the next US president (the one who takes office in 2021) will be able to appoint 0, 1, 2, 3 and more than 3 new judges if (s)he stays in office for (i) four and (ii) eight years. 


# Seismicity before and after hydraulic fracturing in the Horn River Basin, northeast British Columbia 

Consult the article _Investigation of regional seismicity before and after hydraulic fracturing in the Horn River Basin, northeast British Columbia_ by Farahbod et al. (2014). 

a) Using just the year 2006 data in the first four columns of Table 2, calculate separate event rates for the periods when hydraulic fracturing  (i) was and (ii) was not in operation.
Express each rate as the number of events per year, and accompany it with a 95% CI.

b) Repeat the calculations for the individual years 2007 to 2011.

c) Repeat the  calculations for the entire period 2006 to 2011. 

d) Store your results in a tidy `data.frame`.

d) Are you comfortable using Poisson-based CIs for the entire period 2006 to 2011? i.e. does it look like the rate in non-HF days is homogeneous over the 6 years, i.e., do the 6 CIs largely overlap each other?

e) Does it look like the rate in HF days is homogeneous over the 6 years, i.e., do the 6 CIs largely overlap each other?




# De-novo Mutation Rates

The \href{https://www.cancer.gov/publications/dictionaries/genetics-dictionary}
{US NCI Dictionary of Genetics Terms} defines  a \textit{de novo} mutation as 

> a genetic alteration that is present for the first time in one family member as a result of a variant (or mutation) in a germ cell (egg or sperm) of one of the parents, or a variant that arises in the fertilized egg itself during early embryogenesis.

These point mutations differ from the chromosome-level errors that lead to Down Syndrome, and  so the influences of the two parental ages might also be expected to differ. 

The article _Rate of de novo mutations and the importance of fatherâ€™s age to disease risk_ is a good introduction to the fast-moving research on this. The  written news item \href{http://www.nature.com/news/fathers-bequeath-more-mutations-as-they-age-1.11247}{(Nature News, 2012)},  and the audio interview with the senior author \href{https://www.nature.com/nature/podcast/index-2012-08-23.html}{(Stefansson, 2012)} provide a good lay introduction. That study  involved genetic material from  78 father-mother-offspring trios. The focus was on single nucleotide polymorphism mutations and limited to the autosomal chromosomes 1-22, thereby  excluding the sex chromosome 23. Using several rigorous criteria, a total of 4,933 \textit{de novo} mutations, or an average of 63.2 per trio, were called.





a) In five trios, a child of the offspring was also sequenced, thereby allowing the investigators to  determine the parent of origin of each of the 348 \textit{de novo} mutations called. Using Table 1, calculate separate mutation rates for mutations from mothers and from fathers. Express each rate as the number of mutations per parent-year-lived, and accompany it with a 95% CI.














